python/paddle/fluid/layers/nn.py

```diff
@@ -803,10 +846,30 @@ def dropout(x,
             x = fluid.data(name="data", shape=[None, 32, 32], dtype="float32")
             droped = fluid.layers.dropout(x, dropout_prob=0.5)
     """
+    if in_dygraph_mode():
+        if (seed is None or
+                seed == 0) and default_main_program().random_seed != 0:
+            seed = default_main_program().random_seed
+        attrs = {
+            'dropout_prob': dropout_prob,
+            'is_test': is_test,
+            'fix_seed': seed is not None,
+            'seed': seed if seed is not None else 0,
+            'dropout_implementation': dropout_implementation,
+        }
+        if _dygraph_tracer()._train_mode == False:
+            # eval mode
+            attrs['is_test'] = True
+        else:
+            attrs['is_test'] = False
```

Comment:
`attrs['is_test'] = not _dygraph_tracer()._train_mode` ?

Reply:
Yes, thanks.